tests/cnf/ran/ztp/tests/ztp-spoke-checker.go

```diff
@@ -6,40 +6,66 @@ import (
 	"github.com/golang/glog"
 	. "github.com/onsi/ginkgo/v2"
 	. "github.com/onsi/gomega"
+	"github.com/openshift-kni/eco-goinfra/pkg/nodes"
 	"github.com/openshift-kni/eco-goinfra/pkg/pod"
 	"github.com/openshift-kni/eco-goinfra/pkg/reportxml"
 	"github.com/openshift-kni/eco-gotests/tests/cnf/ran/internal/cluster"
 	. "github.com/openshift-kni/eco-gotests/tests/cnf/ran/internal/raninittools"
 	"github.com/openshift-kni/eco-gotests/tests/cnf/ran/ztp/internal/tsparams"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/fields"
 )
 
 var _ = Describe("ZTP Spoke Checker Tests", Label(tsparams.LabelSpokeCheckerTests), func() {
+	var nodeSelector metav1.ListOptions
+
+	BeforeEach(func() {
+		nodeList, err := nodes.List(Spoke1APIClient)
+		Expect(err).ToNot(HaveOccurred(), "Failed to list nodes for spoke 1")
+		Expect(nodeList).ToNot(BeEmpty(), "Failed to find any node for spoke 1")
+
+		nodeSelector = metav1.ListOptions{
+			FieldSelector: fields.SelectorFromSet(fields.Set{"metadata.name": nodeList[0].Definition.Name}).String(),
+		}
+	})
+
 	When("a TunedPerformancePatch.yaml PGT disables the chronyd service", func() {
 		// 54237 - Moving chronyd disable to tuned patch
 		It("should disable and inactivate the chronyd service on spoke", reportxml.ID("54237"), func() {
 			By("verifying chronyd is inactive")
 			// Use `| cat -` to explicitly ignore the return code since it will be nonzero when the service
 			// is not active, which is expected here.
-			status, err := cluster.ExecCommandOnSNO(
-				Spoke1APIClient, 3, "systemctl is-active chronyd.service | cat -")
+			statuses, err := cluster.ExecCmdWithStdout(
+				Spoke1APIClient, 3, "systemctl is-active chronyd.service | cat -", nodeSelector)
```

Comment:
We can probably select all the nodes with worker role. So it doesn't need additional loop other than defining the nodeSelector properly.